test/specs/apexLspRestart.e2e.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { TestSetup } from '../testSetup';
+import * as utilities from '../utilities/index';
+import { EnvironmentSettings } from '../environmentSettings';
+import { By, WebElement, after } from 'vscode-extension-tester';
+import path from 'path';
+import { step } from 'mocha-steps';
+import { expect, util } from 'chai';
+import fs from 'fs';
+
+// Constants
+const projectPath = path.join(__dirname, '..', '..', 'e2e-temp','TempProject-ApexLspRestart');
+const apexClassPath = path.join(projectPath, 'force-app', 'main', 'default', 'classes');
+const INDEXING_COMPLETE = 'Indexing complete';
+const LSP_RESTARTING = 'Apex Language Server is restarting';
+const PRELUDE_STARTING = 'Apex Prelude Service STARTING';
+
+// Helper Functions
+const findReleaseDir = (): string => {
+  const toolsPath = path.join(projectPath, '.sfdx', 'tools');
+  const entries = fs.readdirSync(toolsPath);
+  const match = entries.find(entry => /^\d{3}$/.test(entry));
+  return match || '254';
```

Comment:
a minor code smell. defaults to 254 if we find no match which would still fail at a later point but I'm not sure if that's going to matter much long term.